distributed/tests/test_client.py

```diff
@@ -3599,6 +3599,38 @@ async def test_bad_tasks_fail(c, s, a, b):
     await asyncio.gather(a.close(), b.close())
 
 
+@gen_cluster(
+    client=True,
+    Worker=Nanny,
+    clean_kwargs={"threads": False, "processes": False},
+)
+async def test_suspicious_workers(c, s, a, b):
+    """
+    c: Client
+    s: Scheduler
+    a: Worker # name: 0
+    b: Worker # name: 1
+    """
+    # Create a future on a worker that will fail, on worker a
+    f = c.submit(sys.exit, 0)  # f: Future
+    # logger is a context manager that captures logs i.e. the errors that we will inspect.
+    with captured_logger("distributed.scheduler") as logger:
+        # when we await the future, it will fail and raise a KilledWorker error
+        with pytest.raises(KilledWorker) as info:
+            await f  # wait for the result of the future, which will fail.
+
+    logger_text = logger.getvalue()  # get the logs from the logger
+    assert f.key in logger_text  # check that the key of the future is in the logs
+
+    # look at the client events
+    # look at the suspicious-tasks events
+    # look at the last event
+    # look at the second element of the tuple which is a dictionary
+    # look at the count of the last event and check it is 4 as we expect
```

Comment:
I'm fine keeping this in but generally speaking we do not need verbose comments like this. Mostly the comments are just explaining what the code is already expressing so it is not necessary.
It probably feels useful for you right now because all of this is still new but once you've done this once or twice, these comments won't be that relevant any longer

Reply:
agreed I was planning on removing the comments before converting from draft.